core/src/main/java/io/questdb/cairo/wal/ApplyWal2TableJob.java

```diff
@@ -568,7 +568,8 @@ private int processWalCommit(
                                         txnDetails.getMatViewRefreshTxn(s),
                                         txnDetails.getMatViewRefreshTimestamp(s),
                                         false,
-                                        null
+                                        null,
+                                        txnDetails.getMatViewPeriodHi(s)
```

Comment:
This loop goes from the highest applied `seqTxn` to the lowest. I would expect you want to update this value from the highest applied txn and not overwrite it from the lowest.

Reply:
There is a `break` statement on L580. Added a clarifying comment to make it more visible in bab04d4c398d40312472e41ef2b32cde21c4f6a5

Reply:
What if the last `MAT_VIEW_DATA` has periodHi set to null?

The scenario is that if there were 2 commits in one incremental refresh and another 2 in the second incremental refresh. Then WAL Apply takes the first 3 out of these 4 transactions in the first go. The 3rd transaction is not the final refresh commit and will have periodHi set to null.

Reply:
> What if the last `MAT_VIEW_DATA` has periodHi set to null?

`periodHi` can't jump from a non-null value to a null in mat view data transactions. If it's not set, it's always taken from the state in `MatViewRefreshJob#insertAsSelect()`:
```java
final long commitPeriodHi = periodHi != Numbers.LONG_NULL ? periodHi : viewState.getLastPeriodHi();
```

Same story as with the base table txn.

Reply:
`MAT_VIEW_DATA` is a confusing name. Mat views can have both `DATA` and `MAT_VIEW_DATA`, and the names do not make the difference clear.